common/src/test/java/io/netty/util/internal/logging/Log4J2LoggerTest.java

```diff
@@ -15,211 +15,85 @@
-        Logger mock = mock(Logger.class);
-
-        when(mock.getName()).thenReturn("foo");
-
-        InternalLogger logger = new Log4J2Logger(mock);
-        logger.info("a");
-
-        verify(mock).getName();
-        verify(mock).info("a");
-    }
-
-    @Test
-    public void testInfoWithException() {
-        Logger mock = mock(Logger.class);
-
-        when(mock.getName()).thenReturn("foo");
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.Marker;
+import org.apache.logging.log4j.message.Message;
+import org.junit.Test;
 
-        InternalLogger logger = new Log4J2Logger(mock);
-        logger.info("a", e);
+import io.netty.util.internal.ReflectionUtil;
 
-        verify(mock).getName();
-        verify(mock).info("a", e);
+/**
+ * {@linkplain Log4J2Logger} extends {@linkplain ExtendedLoggerWrapper} implements {@linkplain InternalLogger}.<br>
+ * {@linkplain ExtendedLoggerWrapper} is Log4j2 wrapper class to support wrapped loggers,
+ * so There is no need to test it's method.<br>
+ * We only need to test the netty's {@linkplain InternalLogger} interface method.<br>
+ * It's meaning that we only need to test the Override method in the {@linkplain Log4J2Logger}.
+ */
+public class Log4J2LoggerTest extends AbstractInternalLoggerTest<Logger> {
+
+    {
+        mockLog = LogManager.getLogger(loggerName);
+        logger = new Log4J2Logger(mockLog) {
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public void logMessage(String fqcn, Level level, Marker marker, Message message, Throwable t) {
+                result.put("level", level.name());
+                result.put("t", t);
+                super.logMessage(fqcn, level, marker, message, t);
+            };
+        };
     }
 
     @Test
-    public void testWarn() {
-        Logger mock = mock(Logger.class);
-
-        when(mock.getName()).thenReturn("foo");
-
-        InternalLogger logger = new Log4J2Logger(mock);
-        logger.warn("a");
-
-        verify(mock).getName();
-        verify(mock).warn("a");
+    public void testEXCEPTION_MESSAGE() {
+        assertEquals(getFieldValue(AbstractInternalLogger.class, "EXCEPTION_MESSAGE"),
+                getFieldValue(Log4J2Logger.class, "EXCEPTION_MESSAGE"));
     }
 
-    @Test
-    public void testWarnWithException() {
-        Logger mock = mock(Logger.class);
-
-        when(mock.getName()).thenReturn("foo");
-
-        InternalLogger logger = new Log4J2Logger(mock);
-        logger.warn("a", e);
-
-        verify(mock).getName();
-        verify(mock).warn("a", e);
+    @SuppressWarnings("unchecked")
+    private static <T> T getFieldValue(Class<?> clazz, String fieldName) {
+        try {
+            Field field = clazz.getDeclaredField(fieldName);
+            if (!field.isAccessible()) {
+                assertNull(ReflectionUtil.trySetAccessible(field));
```

Comment:
should use `assumeNull` as it will fail on java9

Reply:
I think it is just running in junit test, so the change code it is ok?
please help me to check:
`assertNull(ReflectionUtil.trySetAccessible(field));` <-- `ReflectionUtil.trySetAccessible(field);`

Reply:
@takeseem change the line to 
```java
assumeNull(ReflectionUtil.trySetAccessible(field));
```

Reply:
@johnou Thanks.
- I first use assumeNull, I found the Assume in junit, no Assume.assumeNull, but exist Assume.assumeNotNull.
- I search `assumeNull` not find in netty project. but I found [Http2FrameCodecTest.upgradeEventNoRefCntError()](https://github.com/netty/netty/blob/4.1/codec-http2/src/test/java/io/netty/handler/codec/http2/Http2FrameCodecTest.java#L661): `Assume.assumeTrue(ReflectionUtil.trySetAccessible(constructor) == null);`

I will use `Assume.assumeThat(ReflectionUtil.trySetAccessible(field), CoreMatchers.nullValue());`, is that good?

Reply:
@johnou @normanmaurer 
- about ReflectionUtil.trySetAccessible
```
    public static Throwable trySetAccessible(AccessibleObject object) {
        try {
            object.setAccessible(true);
            return null;
        } catch (SecurityException e) {
            return e;
        } catch (RuntimeException e) {
            return handleInaccessibleObjectException(e);
        }
    }
```

- I think it's better use AccessController.doPrivileged
```
    public static Throwable trySetAccessible(final AccessibleObject object) {
        try {
            if (!object.isAccessible()) {
                if (System.getSecurityManager() == null) {
                    object.setAccessible(true);
                } else {
                    AccessController.doPrivileged(new PrivilegedAction<Object>() {
                        @Override
                        public Object run() {
                            object.setAccessible(true);
                            return null;
                        }
                        
                    });
                }
            }
            return null;
        } catch (SecurityException e) {
            return e;
        } catch (RuntimeException e) {
            return handleInaccessibleObjectException(e);
        }
    }
```